python/dgl/data/wikics.py

```diff
@@ -0,0 +1,147 @@
+"""Wiki-CS Dataset"""
+import itertools
+import os
+import json
+import numpy as np
+from .. import backend as F
+from ..convert import graph
+from .dgl_dataset import DGLBuiltinDataset
+from ..transforms import to_bidirected, reorder_graph
+from .utils import generate_mask_tensor, load_graphs, save_graphs, _get_dgl_url
+
+
+class WikiCSDataset(DGLBuiltinDataset):
+    r"""Wiki-CS is a Wikipedia-based dataset for node classification.
+
+    The dataset consists of nodes corresponding to Computer Science articles, with edges based on
+    hyperlinks and 10 classes representing different branches of the field.
+
+    Reference: `<https://arxiv.org/abs/2109.01116>`_
+
+    WikiCS dataset statistics:
+
+    - Nodes: 11,701
+    - Edges: 431,726 (note that the original dataset has 216,123 edges but DGL adds
+      the reverse edges and remove the duplicates, hence with a different number)
+    - Number of Classes: 10
+    - Node feature size:: 300
+    - Number of different train, validation, stopping splits: 20
```

Comment:
It seems that you also have a test subset. What does "stopping" stand for here?

Reply:
In the paper, the authors suggest using the validation set for hyperparameter tuning, the stop set for early stopping criterion, and the test set for model evaluation.

Reply:
Is this introduced in "https://arxiv.org/abs/2109.01116"? Does the training set, validation set, test set, and stop set form a partition? Or is one set a subset of another set?

Reply:
This is introduced in "https://arxiv.org/abs/2007.02901v2". They split the nodes in each class into two sets, 50% for the test set and 50% potentially visible. From the visible set, they generated 20 different splits of training, validation and early-stopping sets: 5% of the nodes in each class were used for training in each split, 22.5% were used to evaluate the early-stopping criterion, and 22.5% were used as the validation set for hyperparameter tuning.

Reply:
What's the shape of `g.ndata['train_mask']`? Are these random splits now deterministic?

Reply:
I see. Then perhaps you need to add a description of the shape of these masks and what they stand for.

Reply:
Got it